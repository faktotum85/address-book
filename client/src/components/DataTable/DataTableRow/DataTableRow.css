.actions > * {
    margin: 0 20px;     
    font-weight: bold;
    cursor: pointer;
}

.actions > *:hover {
    text-decoration: underline;
}

.actions > button {
    background: none;
    font-size: inherit;
    border: none;
}

.actions > a {
    text-decoration: none;
    color: inherit;
}