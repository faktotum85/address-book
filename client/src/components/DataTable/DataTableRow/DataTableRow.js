import React from 'react';
import { Link, withRouter } from 'react-router-dom';

import classes from './DataTableRow.css';

const DataTableRow = (props) => {

    const rowData = props.rowData ?
        props.rowData.map((column, i) => <td key={i}>{column}</td>)
        : null;
    const editLink = props.match && props.id !== undefined ? 
        <Link to={props.match.url + '/' + props.id}>
            Edit
        </Link>
        : null;
    const deleteButton = props.deleteHandler ?
        <button onClick={() => props.deleteHandler(props.id)}>
            Delete
        </button>
        : null;
    const actions = editLink || deleteButton ? 
        <td className={classes.actions}>
            {editLink}
            {deleteButton}
        </td>
        : null;
    
    return (
        <tr>
            {rowData}
            {actions}
        </tr>
    );
};

export default withRouter(DataTableRow);
export { DataTableRow };