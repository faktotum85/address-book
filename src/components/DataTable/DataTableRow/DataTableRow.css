.actions {
    text-align: right;
}

.actions > * {
    margin: 0 15px;     
    font-weight: bold;
    cursor: pointer;
}

.actions > *:hover {
    text-decoration: underline;
}

.actions > button {
    background: none;
    font-size: inherit;
    border: none;
    padding: 0;
}

.actions > a {
    text-decoration: none;
    color: inherit;
}