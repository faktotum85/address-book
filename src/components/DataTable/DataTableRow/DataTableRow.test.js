import React from 'react';
import { shallow } from 'enzyme';

import { DataTableRow } from './DataTableRow';

describe('DataTableRow', () => {
    let shallowDataTableRow;
    let props;
    const dataTableRow = (props) => {
        if (!shallowDataTableRow) {
            shallowDataTableRow = shallow(
                <DataTableRow {...props} />
            );
            return shallowDataTableRow;
        }
    }

    beforeEach(() => {
        shallowDataTableRow = undefined;
        props = {
            rowData: undefined,
            deleteHandler: undefined,
            match: undefined,
            id: undefined,
        }
    });

    it('renders correctly', () => {
        const shallowRow = dataTableRow(props);
        expect(shallowRow).toMatchSnapshot();
    });

    it('renders a td per rowData element passed in', () => {
        props.rowData = ['Simon', 'Weller'];
        const shallowRow = dataTableRow(props);
        expect(shallowRow.find('td').length).toBe(2);
    });

    it('renders a delete button in the action cell if a deleteHandler is passed in', () => {
        props.deleteHandler = () => {};
        const shallowRow = dataTableRow(props);
        expect(shallowRow.find('.actions button').length).toBe(1);
    });

    it('calls the deleteHandler with the id when the delete button is clicked', () => {
        const deleteHandler = jest.fn();
        props.deleteHandler = deleteHandler;
        props.id = 27;
        const shallowRow = dataTableRow(props);
        shallowRow.find('.actions button').simulate('click');
        expect(deleteHandler).toBeCalledWith(27, undefined);
    });

    it('renders an edit link in the action cell if an id is passed in and a url can be matched', () => {
        props.id = 0;
        props.match = {
            url: ''
        }
        const shallowRow = dataTableRow(props);
        expect(shallowRow.find('.actions Link').length).toBe(1);
    });

});