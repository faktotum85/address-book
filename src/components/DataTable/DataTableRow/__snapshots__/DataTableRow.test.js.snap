// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableRow renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataTableRow
    deleteHandler={undefined}
    id={undefined}
    match={undefined}
    rowData={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
